source/sonata_tech.rst

```diff
@@ -597,3 +597,5 @@ The format of the weights file is described below:
   /{population_name}                            section_ids             uint64     Total_seg           Mandatory     For each segment, index of the section to which it belongs. Ordered by section ids and grouped by nodes.
   /{population_name}                            offsets                 uint64     N_nodes + 1         Mandatory     The offset for each node in the data and sec_ids fields
   ============================================= ======================= ========== =================== ============= ==========================================
+
+``node_ids`` dataset have an HDF5 attribute of type string named ``circuit`` that gives the path of the circuit to be used.
```

Comment:
`that gives the path of the circuit to be used.`; in the example file, it has `/path/to/nodes.h5`; this really impacts the portability of moving files around, since one would have to update the electrodes file each time a file is moved.  What is this required for?

Reply:
The idea of this was to avoid using an electrodes_file in a simulation which circuit doesn't match.
Right now we check if it matches otherwise we dont generate a report and throw a warning. It was suggested in this ticket
https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-749

Reply:
Ah, I see.
What happens if the different electrodes require different node ids?

Reply:
I guess the same way the circuit_config needs to be updated, we could also update the electrodes_file associated with that circuit?

Reply:
Normally people wouldn't need to change the `circuit_config.json`, since it works with relative paths.  Moreover, it's changing a text file, which is easy, changing attributes on an h5py is not as trivial.

Reply:
The more I think about it, I'm not sure this path is needed - the `edges.h5` files don't refer to their `nodes.h5`, and the expectation is that if there is a dectable mismatch, then the simulator will raise an error (ie: if the morphology doesn't have enough sections); couldn't we do the same here?

Reply:
Its a couple of lines of code, but i get your point. Maybe we can think where to put the electrodes_file so we have relative paths instead, or see once we encounter the issue in the future what would be the best

Reply:
I think that is more clear and fast to fail when the circuit doesnt match instead of going through the sections/segments and see there is a missmatch. But probably @joseph-tharayil knows better here.

Reply:
indeed, we should not put a path (absolute or not) in the binary. Like other artefact, such as spike file, it is up to the user to make sure it is consistent.

Reply:
Maybe we should leave it as 'metadata'? even if we don't enforce/check during the simulation?
Or shall we remove it altogether?

Reply:
we don't have anything that identifies uniquely a circuit (like a checksum or uuid). So I would leave that out of that MR and remove it.

Reply:
done